table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +148,559 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
+      }
+    }
+
+    for (NamePatternEntry entry : mIgnoreEntry.getList().getIncludedEntries()) {
+      if (entry.isPattern()) {
+        builder.ignore().include().addPattern(entry.getPattern());
+      } else {
+        builder.ignore().include().addName(entry.getName());
+      }
+    }
+
+    for (NamePatternEntry entry : mIgnoreEntry.getList().getExcludedEntries()) {
+      if (entry.isPattern()) {
+        builder.ignore().exclude().addPattern(entry.getPattern());
+      } else {
+        builder.ignore().exclude().addName(entry.getName());
+      }
+    }
+    return builder.build();
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    } else if (other == null) {
+      return false;
+    } else if (getClass() != other.getClass()) {
+      return false;
+    }
+    DbConfig that = (DbConfig) other;
+    return Objects.equals(mBypassEntry, that.mBypassEntry)
+        && Objects.equals(mIgnoreEntry, that.mIgnoreEntry);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(mBypassEntry, mIgnoreEntry);
+  }
+
+  /**
+   * Type alias for TablesEntry<TableEntry>.
+   */
+  @JsonDeserialize(using = BypassTablesSpecDeserializer.class)
+  static final class BypassTablesSpec extends TablesEntry<TableEntry> {
+    // inherited: IncludeExcludeList<TableEntry>
+    // this is the BypassTablesObject from syntax specification
+
+    BypassTablesSpec(@Nullable IncludeExcludeList<TableEntry> list) {
+      super(list);
+    }
+  }
+
+  /**
+   * Type alias for TablesEntry<NamePatternEntry>.
+   */
+  @JsonDeserialize(using = IgnoreTablesSpecDeserializer.class)
+  static final class IgnoreTablesSpec extends TablesEntry<NamePatternEntry> {
+    // inherited: IncludeExcludeList<NamePatternEntry>
+    // this is the SimpleIncludeExcludeObject from syntax specification
+
+    IgnoreTablesSpec(@Nullable IncludeExcludeList<NamePatternEntry> list) {
+      super(list);
+    }
   }
 
   /**
-   * @return the {@link BypassEntry} from config file
+   * The "tables" object: {"tables": ... }.
+   * Base class for BypassTablesSpec and IgnoreTablesSpec.
+   * @param <T> the type of entry contained
    */
-  public BypassEntry getBypassEntry() {
-    return mBypassEntry;
+  static class TablesEntry<T extends NamePatternEntry> {
+    static final String FIELD_TABLES = "tables";
+
+    private final IncludeExcludeList<T> mList;
+
+    protected TablesEntry(@Nullable IncludeExcludeList<T> list) {
+      mList = list == null ? IncludeExcludeList.empty() : list;
+    }
+
+    IncludeExcludeList<T> getList() {
+      return mList;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (this == other) {
+        return true;
+      } else if (other == null) {
+        return false;
+      } else if (getClass() != other.getClass()) {
+        return false;
+      }
+      TablesEntry that = (TablesEntry) other;
+      return Objects.equals(mList, that.mList);
+    }
```

Comment:
Do we still need this? If we need, change to a helper function since we saw this code several times.

Reply:
Extracting the code into a function triggers spotbugs error:

```
[ERROR] Medium: Equals method for alluxio.master.table.DbConfig assumes the argument is of type DbConfig [alluxio.master.table.DbConfig] At DbConfig.java:[line 231] BC_EQUALS_METHOD_SHOULD_WORK_FOR_ALL_OBJECTS
```

Reply:
This is circumvented with `    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")` and `
@SuppressFBWarnings("EQ_UNUSUAL")
`.